config/misc.go

```diff
@@ -35,8 +36,8 @@ func GetInstanceID() string {
 	if instance == "" {
 		return ""
 	}
-	instanceArr := strings.Split(instance, "-")
-	return instanceArr[len(instanceArr)-1]
+	re := regexp.MustCompile("-[0-9]+([A-Za-z0-9-]*)$")
+	return strings.TrimPrefix(string(re.Find([]byte(instance))), "-")
```

Comment:
Don't we just need to find the first segment that represents a number?
```suggestion
	// split instance id by "-" and return the first number
	if instanceID, ok := lo.Find(
		strings.Split(instance, "-"),
		func(s string) bool {
			_, err := strconv.Atoi(s)
			return err == nil
		}); ok {
		return instanceID
	}
	return ""
```

Reply:
We are also making the scheduler changes accordingly to wait for both gateway pods to acknowledge